src/Features/Protocol/LanguageServicesTest/RoslynLanguageServiceTests.cs

```diff
@@ -0,0 +1,924 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Editor.UnitTests.Workspaces;
+using Microsoft.CodeAnalysis.Protocol.LanguageServices.Extensions;
+using Microsoft.CodeAnalysis.Test.Utilities;
+using Microsoft.CodeAnalysis.Text;
+using Microsoft.VisualStudio.LanguageServer.Protocol;
+using Roslyn.Utilities;
+using Xunit;
+using VSSymbolKind = Microsoft.VisualStudio.LanguageServer.Protocol.SymbolKind;
+using VSLocation = Microsoft.VisualStudio.LanguageServer.Protocol.Location;
+
+namespace Microsoft.CodeAnalysis.Protocol.LanguageServices.UnitTests
+{
+    [UseExportProvider]
+    public class RoslynLanguageServiceTests
+    {
+        [Fact]
+        public async Task TestGetDocumentSymbolsAsync__WithHierarchicalSupport()
+        {
+            var markup =
+@"{|class:class A
+{
+    {|method:void M()
+    {
+    }|}
+}|}";
+            var (solution, locations) = CreateTestSolution(markup);
+            var expected = new DocumentSymbol[]
+            {
+                CreateDocumentSymbol(VSSymbolKind.Class, "A", locations["class"].First())
+            };
+            CreateDocumentSymbol(VSSymbolKind.Method, "M", locations["method"].First(), expected.First());
+
+            var results = await RunGetDocumentSymbolsAsync(solution, true);
+            AssertCollection(expected, results, AssertDocumentSymbolEquals);
+        }
+
+        [Fact]
+        public async Task TestGetDocumentSymbolsAsync__WithoutHierarchicalSupport()
+        {
+            var markup =
+@"class {|class:A|}
+{
+    void {|method:M|}()
```

Comment:
When a **SymbolInformation** class is returned rather than a **DocumentSymbol**, the locations returned are also different.  In **SymbolInformation**, we just get for example the method name ```M```.  But in **DocumentSymbol**, we get the entire method block
```csharp
void M()
{
}
```
Which is right, and should they be different? @heejaechang